drivers/backup/portworx/portworx.go

```diff
@@ -252,6 +255,30 @@ func (p *portworx) DeleteBackupSchedule(req *api.BackupScheduleDeleteRequest) (*
 	return p.backupScheduleManager.Delete(context.Background(), req)
 }
 
+func (p *portworx) WaitForRunning(req *api.BackupInspectRequest, timeout, retryInterval time.Duration) error {
+	t := func() (interface{}, bool, error) {
+		resp, err := p.backupManager.Inspect(context.Background(), req)
+
+		if err != nil {
+			return nil, true, err
+		}
+
+		// If backup is in progress - stop
+		if resp.Backup.GetStatus().Status == api.BackupInfo_StatusInfo_InProgress {
+			return "", false, nil
+		}
```

Comment:
@stgleb on other terminal conditions like failed, invalid statuses function should give up checking for backup to move to InProgress state.

Reply:
gotcha thanks

Reply:
Done

Reply:
@stgleb did you test these changes? With my runs, `task.DoRetryWithTimeout()` func is not returning error in case of failed, aborted, invalid cases, so I am explicitly returning  error from `WaitFor...()` function

Reply:
@stgleb once we do above change, we should be good to merge this.